imports/plugins/core/orders/client/components/OrderRefunds.js

```diff
@@ -0,0 +1,303 @@
+
+const styles = (theme) => ({
+  dividerSpacing: {
+    marginBottom: theme.spacing(4),
+    marginTop: theme.spacing(4)
+  },
+  fontWeightSemiBold: {
+    fontWeight: theme.typography.fontWeightSemiBold
+  },
+  formControl: {
+    minWidth: "250px",
+    width: "250px"
+  },
+  selectEmpty: {
+    marginTop: theme.spacing(2)
+  }
+});
+
+/**
+ * @name OrderRefunds
+ * @param {Object} props Component props
+ * @returns {React.Component} returns a React component
+ */
+function OrderRefunds(props) {
+  const hasPermission = Reaction.hasPermission(["reaction-orders", "order/fulfillment"], Reaction.getUserId(), Reaction.getShopId());
+  const { classes, order } = props;
+  const { payments } = order;
+
+  // useRef
+  const inputLabel = useRef(null);
+
+  // useState
+  const [labelWidth, setLabelWidth] = useState(0);
+  const [refundReasonSelectValues, setRefundReasonSelectValues] = useState({ reason: "" });
+  const [refundTotal, setRefundTotal] = useState(0.00);
+
+  // useEffect
+  // update label width when refund select is activate
+  useEffect(() => {
+    if (inputLabel && inputLabel.current) {
+      setLabelWidth(inputLabel.current.offsetWidth);
+    }
+  }, []);
+
+  const canRefund = payments.every((payment) => payment.status !== "refunded");
+  // calculated refund total from inputs
+  const calculatedRefundTotalDisplay = formatMoney(refundTotal, order.currencyCode);
+  // previous refunds
+  const orderPreviousRefundTotal = order.refunds.reduce((acc, refund) => acc + refund.amount.amount, 0);
+  // available to refund
+  const orderAmountAvailableForRefund = order.summary.total.amount - orderPreviousRefundTotal;
+  const orderAmountAvailableForRefundDisplay = formatMoney(orderAmountAvailableForRefund, order.currencyCode);
+
+  const handleCreateRefund = (data, mutation) => {
+    const { amounts } = data;
+    const { reason } = refundReasonSelectValues;
+
+    // turn form data into an array of payments that provide paymentID and amount
+    // then filter out any amounts that are `null` or `0`
+    const paymentsToRefund = Object.keys(amounts).map((paymentId) => ({
+      paymentId,
+      amount: parseFloat(amounts[paymentId], 10)
+    })).filter((payment) => payment.amount && payment.amount > 0);
+
+    paymentsToRefund.forEach((payment) => {
+      const variables = {
+        amount: payment.amount,
+        orderId: order._id,
+        paymentId: payment.paymentId
+      };
+
+      // Stripe will not accept an empty string or `null` value for the `reason` field,
+      // so only include `reason` in the mutation only if its' value is set
+      if (reason) {
+        variables.reason = reason;
+      }
+
+      if (hasPermission) {
+        mutation({
+          variables
+        });
+      }
+    });
+  };
+
+  // When refund amounts are changed, add up amounts to display in button
+  const handleRefundTotalUpdate = () => {
+    if (this.form && this.form.state) {
```

Comment:
It might be simpler and cleaner to implement this as an `onChange` or `onChanging` prop on `Form`?

Reply:
@aldeed It's kind of already doing that. I might be doing one extra step by calling `handleRefundAmountChange` on `onChanging`, which in turn calls `handleRefundTotalUpdate`. Is there something else that should be changed aside from maybe removing the extra step?